### ISSUE TITLE
Problem with unfold and sketch on sheet metal functions

### ISSUE THREAD

**Issue by Pendelf** [shaise/FreeCAD_SheetMetal#272]
https://youtu.be/UFDPVvr0Jbw

**Comment by luzpaz** [comment]
Missing full [About](https://wiki.freecad.org/About) info.

**Comment by Pendelf** [comment]
OS: Manjaro Linux (KDE/plasmawayland)
Word size of FreeCAD: 64-bit
Version: 2023.131.26244 +5365 (Git) AppImage
Build type: Release
Branch: LinkDaily
Hash: 73c4ca254c5de8849d104973482e98399fb24863
Python version: 3.10.8
Qt version: 5.15.4
Coin version: 4.0.1
OCC version: 7.6.3
Locale: English/United States (en_US)

**Comment by Pendelf** [comment]
https://youtu.be/ANFXfNuRbB8

[code]
OS: Manjaro Linux (KDE/plasmawayland)
Word size of FreeCAD: 64-bit
Version: 0.20.2.29603 (Git)
Build type: Release
Branch: makepkg
Hash: 930dd9a76203a3260b1e6256c70c1c3cad8c5cb8
Python 3.10.9, Qt 5.15.8, Coin 4.0.1, Vtk 9.1.0, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * sheetmetal 0.2.62
  * A2plus 0.4.60k
  * BIM 2021.12.0
  * Curves 0.6.8
  * dodo 1.0.0
  * ExplodedAssembly
  * fasteners 0.4.55
  * LCInterlocking
[/code]

**Comment by Syres916** [comment]
@Pendelf please upgrade to the Wb to version 0.2.63 so we're both using the same version during my testing, some unrelated bugs were fixed.

I'll do some investigation using 0.2.63 but this specific procedure isn't something that I've had to do in real life work so it might take some time.

**Comment by Pendelf** [comment]
I wanted to do something like this. It’s not all 90-degree bends, plus there are slots on these bends. Basically, I got it all but a unfold

![a4q8phcwxopa1](https://user-images.githubusercontent.com/128366440/228603173-0026df2f-f5f7-442d-beff-da48b6d7794f.png)

**Comment by shaise** [comment]
the unfold property is not useful in this case. if you want to unfold the model, just select any flat surface and click the unfold toolbar button.
![image](https://user-images.githubusercontent.com/5813568/228623880-2bc9a5d1-2e43-455d-a834-931a33bd4555.png)

**Comment by Pendelf** [comment]
yeah, that’s what I was doing. I use the unfold property when make a sketch on a face. When a model is finished for unfold i use unfold toolbar button

**Comment by shaise** [comment]
The unfold property only works if there are only fold actions for the model. If you add other actions such as cutting holes, the unfold property will not work.

**Comment by Pendelf** [comment]
Okay, I get it. It wouldn’t be bad if there was such a feature. You do a time unfold, you put on a sketch, you cut it out, and then you bend it back, and then you do a laser unfold at the end

**Comment by Pendelf** [comment]
Issue is closed, but it's not clear if it works or not? Maybe I'm doing something wrong?

https://github.com/user-attachments/assets/c8fea218-c121-480c-a2a8-7c2618755d97

**Comment by shaise** [comment]
@Pendelf , using the unfold property is not the correct way to work with the sheet metal WB. It will not work if you add features such as holes.
There are two ways to work with that:
1. start with a flat plane, create all the holes you need, then add the folds using bend on sketch. To unfold use the Unfold command icon.
2. create sheet metal model, add holes using sketches using the Extruded Cutout command (works better then Sketch on Sheet Metal) then, again, use the Unfold command to unfold. Do not use the unfold property.

See video:
https://youtu.be/oc_epUv382A

**Comment by Pendelf** [comment]
@shaise, In the second case, the rounding on the fold is not correct

![Снимок экрана 2025-01-13 110900](https://github.com/user-attachments/assets/53a72669-ed87-49b3-9a3c-0fcc50ed405b)

**Comment by shaise** [comment]
@Pendelf ,
Of course! because it was originally cut in diagonal. This is used for example if you want to match the sheetmetal to an extruded profile:
![image](https://github.com/user-attachments/assets/515496aa-44b7-4221-a1da-742a22d2e04b)

Otherwise you either need to use option 1, or make 2 sketches one flat on the base and one on the flange, and make 2 separate cuts.

**Comment by Pendelf** [comment]
@shaise Ok, I get it, anyway thanks for the work done and for developing the workbench. I'll attach another video to clarify what I'm talking about. First we create the bends we want, then we temporarily straighten them, then we make the cut we want and bend them back again


https://github.com/user-attachments/assets/50cb6139-9e41-4e8a-b9ef-280f4cdf02b0

**Comment by shaise** [comment]
@Pendelf , Yes, I understood your problem. Its just that this workbench unfortunately does not support that kind of workflow. It may look simple but its mathematically complicated to implement.
